### ISSUE TITLE
Multiple Websockets being created

### ISSUE THREAD

**Issue by alloystory** [GetStream/stream-chat-react-native#978]
repro:
1. open app
2. put app into background
3. put app into foreground
4. repeat 2 and 3

Hi, I recently launched a support ticket mentioning that some of my users have faced a Too many requests error. Diving deeper into the code, I realised that when the app goes from background into foreground, multiple websockets are opened.

Further diving deeper, it seems that the code at https://github.com/GetStream/stream-chat-react-native/blob/9e742476082a80c9b25b42117bbb1d69a2b2b63a/package/src/components/Chat/hooks/useAppStateListener.ts#L7 is not well memoized. Hence, multiple calls to onForeground and onBackground happens

I have patched my own library to fix the issue, but it'd be great if the team can look into the issue for future versions too

```
export const useAppStateListener = (onForeground?: () => void, onBackground?: () => void) => {
  const appState = useRef(AppState.currentState)

  const handleAppStateChange = useCallback((nextAppState: AppStateStatus) => {
    if (appState.current === 'background' && nextAppState === 'active' && onForeground) {
      onForeground();
    } else if (appState.current.match(/active|inactive/) && nextAppState === 'background' && onBackground) {
      onBackground();
    }
    appState.current = nextAppState
  }, [onForeground, onBackground]);

  useEffect(() => {
    AppState.addEventListener('change', handleAppStateChange);

    return () => AppState.removeEventListener('change', handleAppStateChange);
  }, [handleAppStateChange]);
};
```

gz#16454

**Comment by madsroskar** [comment]
Hi @alloystory - thank you for raising this issue, and sharing the result of your investigation of this!

We will have a look at this, and get back to you once we've checked it out on our end and have a fix ready.

I really appreciate the specific information you've added here, that's really helpful!

Thanks,
Mads

**Comment by madsroskar** [comment]
For our internal tracking: Issue created is CRNS-473

**Comment by vishalnarkhede** [comment]
Hey @alloystory this issue has already been fixed as part of [`v3.9.0`](https://github.com/GetStream/stream-chat-react-native/releases/tag/v3.9.0).

**Comment by vishalnarkhede** [comment]
Please let us know if you are facing this issue on latest version of SDK as well. I am closing this issue for now.
